### ISSUE TITLE
Error detected while processing function provider#clipboard#Call with clipboard=unnamed

### ISSUE THREAD

**Issue by rafaeldelboni** [neovim/neovim#15350]
### Neovim Version

```bash
NVIM v0.5.0
Build type: Release
LuaJIT 2.0.5
Compilation: /usr/bin/cc -D_FORTIFY_SOURCE=2 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNVIM_TS_HAS_SET_MATCH_LIMIT -O2 -DNDEBUG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/build/neovim/src/neovim-0.5.0/build/config -I/build/neovim/src/neovim-0.5.0/src -I/usr/include -I/build/neovim/src/neovim-0.5.0/build/src/nvim/auto -I/build/neovim/src/neovim-0.5.0/build/include
Compiled by builduser

Features: +acl +iconv +tui

```

### Operating system/version:

Arch Linux x86_64 / 5.13.9-arch1-1

### Terminal name/version:

Alacritty

### TERM environment variable

tmux-256color

### Installation

https://archlinux.org/packages/community/x86_64/neovim/

### Steps to reproduce

> This is a clipboard related issue, so make sure you are typing the commands instead of copy/pasting.

First open nvim `nvim --clean` and do:
```
 :let @+ = ''
```

So that the + register is empty, then exit and do:
```
nvim --clean -c 'set clipboard=unnamedplus'
:registers
```

Somehow nvim doesn't like that the + register is accessed when it's not been set already.

### Expected behavior

No errors should be thrown.

### Actual behavior

```vim
Type Name Content
  c  "*   ore
clipboard: error: Error: target STRING not available
Error detected while processing function provider#clipboard#Call[6]..2:
line    8:
E691: Can only compare List with List
E15: Invalid expression: match(&clipboard, '\v(unnamed|unnamedplus)') >= 0 && get(s:selections[a:reg].data, 0, []) == clipboard_data
clipboard: provider returned invalid data
Press ENTER or type command to continue
```

Related: https://github.com/mg979/vim-visual-multi/issues/183

**Comment by matveyt** [comment]
The first error is due to `xclip` itself. See #2642, #7958 etc. It just doesn't like if TARGET is missing (e.g. clipboard is empty).

The second error should disappear after updating Neovim runtime.

**Comment by justinmk** [comment]
> fixed by #14984

requires nvim 0.6
